util/logtools/encounter_tools.ts

```diff
@@ -0,0 +1,313 @@
+  }
+  constructor() {
+    this.currentZone = {};
+    this.currentFight = {};
+    this.currentSeal = null;
+    // May be null for some zones.
+    this.zoneInfo = null;
+
+    this.haveWon = false;
+    this.haveSeenSeals = false;
+
+    this.regex = {
+      changeZone: NetRegexes.changeZone({ id: '*' }),
+      cactbotWipe: NetRegexes.echo({ line: 'cactbot wipe.*?' }),
+      win: NetRegexes.network6d({ command: '40000003' }),
+      wipe: NetRegexes.network6d({ command: '40000010' }),
+      commence: NetRegexes.network6d({ command: '4000000[16]' }),
+      playerAttackingMob: NetRegexes.ability({ sourceId: '1.{7}', targetId: '4.{7}' }),
+      mobAttackingPlayer: NetRegexes.ability({ sourceId: '4.{7}', targetId: '1.{7}' }),
+    };
+
+    // TODO: consider also doing this with lang? But lang only deals with Regexes???
+    // Sorry, this is quite a bit of a hack.
+    this.sealRegexes = [];
+    this.unsealRegexes = [];
+
+    const sealReplace = commonReplacement.replaceSync[syncKeys.seal];
+    if (!sealReplace)
+      throw new Error('missing seal regex');
+    for (const regexString of Object.values(sealReplace)) {
+      const line = `${regexString?.replace('$1', '(?<seal>.*?)')}.*?`;
+      this.sealRegexes.push(NetRegexes.message({ line: line }));
+    }
+
+    const unsealReplace = commonReplacement.replaceSync[syncKeys.unseal];
+    if (!unsealReplace)
+      throw new Error('missing unseal regex');
+    for (const lang of Object.values(unsealReplace)) {
+      const line = `*.?${lang}.*?`;
+      this.unsealRegexes.push(NetRegexes.message({ line: line }));
+    }
+  }
+
+  skipZone(): boolean {
+    if (!this?.currentZone?.zoneId || this.zoneInfo === null)
+      return false;
+    const info = this.zoneInfo;
+    if (!(info?.contentType))
+      return false;
+    const content = info?.contentType;
+    if (!content)
+      return false;
+
+    // There are some seal messages in older raids, but not consistently.
+    // Therefore, we can't require them.
+    const keepTypes: Array<number> = [
+      ContentType.Dungeons,
+      ContentType.Eureka,
+      ContentType.Raids,
+      ContentType.Trials,
+      ContentType.UltimateRaids,
+      ContentType.DeepDungeons,
+    ];
+
+    return !keepTypes.includes(content);
+  }
+
+  process(line: string): void {
+    const cZ = this.regex.changeZone.exec(line);
+    if (cZ?.groups) {
+      if (this?.currentZone?.name === cZ.groups.name) {
+        // Zoning into the same zone, possibly a d/c situation.
+        // Don't stop anything?
+        return;
+      }
+
+      // If we changed zones, we have definitely stopped fighting.
+      // Therefore we can safely initialize everything.
+      if (this.currentFight.startTime)
+        this.initializeFight();
+      if (this.currentZone.startTime)
+        this.initializeZone();
+
+      this.haveWon = false;
+      this.haveSeenSeals = false;
+      this.zoneInfo = ZoneInfo[parseInt(cZ.groups.id, 16)] || null;
```

Comment:
```suggestion
      this.zoneInfo = ZoneInfo[parseInt(cZ.groups.id, 16)] ?? null;
```

... or alternatively just `this.zoneInfo = ZoneInfo[parseInt(cZ.groups.id, 16)` if you make it `undefined` instead of `null`.